File: output/deepseek_r1/avatar/algo_based_translation/java/python/failed_translation/atcoder_AGC019_A.py

Q, H, S, D, N = map(int, input().split())
one_price = min(Q * 4, H * 2, S)
if one_price <= D / 2:
    total = N * one_price
else:
    total = (N // 2) * D + (N % 2) * one_price
print(total)

File: output/deepseek_r1/avatar/direct_translation/python/java/atcoder_AGC019_A.java

import java.util.Scanner;

public class atcoder_AGC019_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int q = sc.nextInt();
        int h = sc.nextInt();
        int s = sc.nextInt();
        int d = sc.nextInt();
        int n = sc.nextInt();
        
        int best1L = Math.min(q * 4, Math.min(h * 2, s));
        int best2L = Math.min(d, best1L * 2);
        
        long total;
        if (n % 2 == 0) {
            total = (long) best2L * (n / 2);
        } else {
            total = (long) best2L * (n / 2) + best1L;
        }
        
        System.out.println(total);
    }
}

